### PR TITLE
[feature](Azure) Implement Azure's multi part upload in BE

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Previously the multipart upload of Azure is not supported, this pr makes it available.

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

File: be/src/io/fs/obj_storage_client.h [hunk 1]
```diff
@@ -45,7 +45,12 @@ struct ObjectStoragePathOptions {
     std::optional<std::string> upload_id = std::nullopt; // only used for S3 upload
 };
 
-struct ObjectCompleteMultiParts {};
+struct ObjectCompleteMultiPart {
+    int part_num = 0;
+    std::optional<std::string> etag = std::nullopt;
+    int get_part_num() const { return part_num; }
```
Comment (platoneko) [inline be/src/io/fs/obj_storage_client.h:51]: 不太需要这些 getter

File: be/src/io/fs/azure_obj_storage_client.cpp [hunk 2]
```diff
@@ -38,6 +41,14 @@ std::string wrap_object_storage_path_msg(const doris::io::ObjectStoragePathOptio
     return fmt::format("bucket {}, key {}, prefix {}, path {}", opts.bucket, opts.key, opts.prefix,
                        opts.path.native());
 }
+
+auto base64_encode_part_num(int part_num) {
+    auto part_num_str = std::to_string(part_num);
```
Comment (platoneko) [inline be/src/io/fs/azure_obj_storage_client.cpp:46]: 似乎不太需要 to_string，直接对 int 计算 base64 也行？

**Review by platoneko — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by platoneko — APPROVED** [review]

**Pull request merged**